File: src/connector/Organization.java
package connector;

import com.sap.conn.jco.JCoDestination;
import com.sap.conn.jco.JCoFunction;
import com.sap.conn.jco.JCoParameterList;
import com.sap.conn.jco.JCoStructure;

public class Organization {

	private JCoDestination destination;
	
	public String
	organization,
	division,
	group,
	office;
	
	public Organization(JCoDestination destination)
	{
		this.destination = destination;
		GetOrganizationDetails();
	}
	
	 public void GetOrganizationDetails()
     {	
    	try{
    		JCoFunction f = destination.getRepository().getFunction("BAPI_SALESORG_GET_DETAIL");
    		f.getImportParameterList().setValue("SALESORGANIZATION", "1000");
    		f.execute(destination);
    		
    		JCoParameterList exportParameters = f.getExportParameterList();
    		JCoStructure info = exportParameters.getStructure("SDORGDATA");
    		
    		System.out.println(exportParameters.getStructure("RETURN").getString("TYPE"));
    		System.out.println(exportParameters.getStructure("RETURN").getString("MESSAGE"));
    		
    		
    		organization = info.getString("SALESORG_TEXT");
    		division = info.getString("DIVISION_TEXT");
    		group = info.getString("SALES_GRP_TEXT");
    		office = info.getString("SALES_OFF_TEXT");
    	} catch(Exception e){}
     }
	
	public String toString()
	{
		return
    			Utility.EmptyOr(organization) + 
    			Utility.EmptyOr(division) + 
    			Utility.EmptyOr(group) +
    			Utility.EmptyOr(office);		
	}
}

File: ERP_WebInterface/ERP_WebInterface/Organization.cs
using SAP.Middleware.Connector;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ERP_WebInterface
{
    public class Organization
    {
        private RfcDestination destination;

        public String
        organization,
        division,
        group,
        office;

        public Organization(RfcDestination destination)
        {
            this.destination = destination;
            GetOrganizationDetails();
        }

        public void GetOrganizationDetails()
        {
            try
            {
                var f = destination.Repository.CreateFunction("BAPI_SALESORG_GET_DETAIL");
                f.SetValue("SALESORGANIZATION", "1000");
                f.Invoke(destination);
                var info = f.GetStructure("SDORGDATA");
                Console.WriteLine(f.GetStructure("RETURN").GetString("TYPE"));
                Console.WriteLine(f.GetStructure("RETURN").GetString("MESSAGE"));
                organization = info.GetString("SALESORG_TEXT") ?? "";
                division = info.GetString("DIVISION_TEXT") ?? "";
                group = info.GetString("SALES_GRP_TEXT") ?? "";
                office = info.GetString("SALES_OFF_TEXT") ?? "";
            }
            catch (Exception e) { }
        }

        public override String ToString()
        {
            return $"{organization}\n{division}\n{group}\n{office}";
        }
    }
}
